### PR TITLE
feat: SelectField component

### PR DESCRIPTION
**Pull request by Heleen-cr**
fixes: #116

### CODE REVIEW

File: src/components/SelectField/SelectField.js [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+    });
+    this.validateField(e);
+  }
+
+  /**
+   * Get the default option, marked as selected from the props.options array
+   * Returns the value or undefined
+   */
+  getSelectedOption() {
+    let selected = this.props.options.find(item => item.selected === true);
```
Comment (deniselashlley) [inline src/components/SelectField/SelectField.js]: check if find() is support by IE10/IE9 sure there was some issue with certain version using find()

File: src/stories/index.js [hunk 2]
```diff
@@ -263,14 +264,33 @@ storiesOf('Input Field', module)
       />);
     }),
   );
+const optionsArray = [
+  { label: 'Please select', value: 'PLEase Select' },
```
Comment (deniselashlley) [inline src/stories/index.js]: typo capital 'E'

File: src/components/SelectField/SelectField.js [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+        valid: true,
+        message: '',
+        value,
+        showErrorMessage: false,
+      });
+    }
+  }
+
+  render() {
+    const errorClass = this.state.showErrorMessage === true && 'form__field-error-wrapper';
```
Comment (deniselashlley) [inline src/components/SelectField/SelectField.js]: removing the : ' ', you will end with a false value in a class

File: src/components/SelectField/SelectField.js [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+      >{item.label}</option>));
+    return options;
+  }
+
+  /**
+   * Validate the select field and update the state with validation info
+   * @param e
+   */
+  validateField(e) {
+    let value = e ? e.target.value : this.inputRef.value;
```
Comment (deniselashlley) [inline src/components/SelectField/SelectField.js]: Does this work?  checking against just e? should it not be !== null

File: src/components/SelectField/SelectField.js [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+        showErrorMessage: false,
+      });
+    }
+  }
+
+  render() {
+    const errorClass = this.state.showErrorMessage === true && 'form__field-error-wrapper';
+    const extraClass = this.props.extraClass !== '' && this.props.extraClass;
+    return (
+      <div id={`field-wrapper--${this.props.id}`} className={`form__fieldset form__field--wrapper form__field-wrapper--select ${errorClass} ${extraClass}`}>
```
Comment (AndyEPhipps) [inline src/components/SelectField/SelectField.js]: yaarp, break all of these out into separate lines pls

File: src/components/SelectField/SelectField.js (hunk 1 above)
Comment (Heleen-cr) [reply src/components/SelectField/SelectField.js]: good point 👍

**Pull request merged**